crates/text/src/lib.rs

```diff
@@ -0,0 +1,365 @@
+            ..Default::default()
+        }
+    }
+
+    pub fn translation<T: Into<Cow<'a, str>>>(translate: T, with: Vec<Text<'a>>) -> Self {
+        Text {
+            content: Content::translation(translate, with),
+            ..Default::default()
+        }
+    }
+
+    pub fn extra(mut self, text: Text<'a>) -> Self {
+        self.extra.push(text);
+        self
+    }
+
+    pub fn color(mut self, color: Color) -> Self {
+        self.color = Some(color);
+        self
+    }
+
+    pub fn clear_color(mut self) -> Self {
+        self.color = None;
+        self
+    }
+
+    pub fn style(mut self, style: Style) -> Self {
+        match style {
+            Style::Obfuscated => self.obfuscated = Some(true),
+            Style::Bold => self.bold = Some(true),
+            Style::Strikethrough => self.strikethrough = Some(true),
+            Style::Underlined => self.underlined = Some(true),
+            Style::Italic => self.italic = Some(true),
+        }
+        self
+    }
+
+    pub fn not_style(mut self, style: Style) -> Self {
+        match style {
+            Style::Obfuscated => self.obfuscated = Some(false),
+            Style::Bold => self.bold = Some(false),
+            Style::Strikethrough => self.strikethrough = Some(false),
+            Style::Underlined => self.underlined = Some(false),
+            Style::Italic => self.italic = Some(false),
+        }
+        self
+    }
+
+    // TODO: Text::style_clear_all
+    pub fn clear_style(mut self, style: Style) -> Self {
+        match style {
+            Style::Obfuscated => self.obfuscated = None,
+            Style::Bold => self.bold = None,
+            Style::Strikethrough => self.strikethrough = None,
+            Style::Underlined => self.underlined = None,
+            Style::Italic => self.italic = None,
+        }
+        self
+    }
+
+    pub fn insertion<S: Into<Cow<'a, str>>>(mut self, value: S) -> Self {
+        let value = value.into();
+        // TODO: is there a difference betwteen None and Some("")?
+        if value.is_empty() {
+            self.insertion = None;
+        } else {
+            self.insertion = Some(value);
+        }
+        self
+    }
+
+    pub fn on_click(mut self, click_event: ClickEvent<'a>) -> Self {
+        self.click_event = Some(click_event);
+        self
+    }
+
+    pub fn clear_on_click(mut self) -> Self {
+        self.click_event = None;
+        self
+    }
+
+    pub fn append(self, text: Text<'a>) -> Self {
+        Text::of("").extra(self).extra(text)
+    }
+
+    pub fn has_color(&self) -> bool {
+        self.color.is_none()
+    }
+
+    pub fn has_style(&self) -> bool {
+        self.bold.is_some()
+            && self.italic.is_some()
+            && self.obfuscated.is_some()
+            && self.strikethrough.is_some()
+            && self.underlined.is_some()
+    }
+
+    // is_bold? Bold can either be None, Some(false), and Some(true).
+    // Maybe a has_bold instead? self.bold.is_some()
```

Comment:
I think each field of `Text` should be made private, and then we expose getter methods that abstract away this implementation detail.